### DISCUSSION TITLE
Initial Chain Switch

### DISCUSSION THREAD

**Discussion by mike-lush** [wevm/wagmi#3826] [Q&A]
Hi,

I have a multi-chain dapp that works on Mainnet and Base.

I have the following issue:

When users connect their wallet to the dapp they are automatically asked to switch to whichever chain is first on the chains list in the config. This is not great for my UI as users who chose Base => Connect Wallet are asked to connect to Mainnet first and then switch back.

How can I disable the initial chain switch request upon connecting wallet.

**Comment by jaredvu** [comment]
Hi @mike-lush, was your problem ever resolved?

**Reply by tmm** [reply]
> When users connect their wallet to the dapp they are automatically asked to switch to whichever chain is first on the chains list

This shouldn't be the case unless you explicitly pass `chainId` to `connect`. Mind sharing some code?

**Reply by jaredvu** [reply]
I think I tracked my issue to this being a feature in Family’s Connectkit.
